File: app/src/main/java/xyz/sleekstats/completist/model/GenreList.java
package xyz.sleekstats.completist.model;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class GenreList implements Parcelable {

    @SerializedName("genres")
    private final List<Genre> genres;

    public List<Genre> getGenres() {
        return genres;
    }

    private GenreList(Parcel in) {
        if (in.readByte() == 0x01) {
            genres = new ArrayList<>();
            in.readList(genres, Genre.class.getClassLoader());
        } else {
            genres = null;
        }
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        if (genres == null) {
            dest.writeByte((byte) (0x00));
        } else {
            dest.writeByte((byte) (0x01));
            dest.writeList(genres);
        }
    }

    @SuppressWarnings("unused")
    public static final Parcelable.Creator<GenreList> CREATOR = new Parcelable.Creator<GenreList>() {
        @Override
        public GenreList createFromParcel(Parcel in) {
            return new GenreList(in);
        }

        @Override
        public GenreList[] newArray(int size) {
            return new GenreList[size];
        }
    };
}

File: app/src/main/java/xyz/sleekstats/completist/model/GenreList.kt
package xyz.sleekstats.completist.model

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import java.util.*

class GenreList private constructor(`in`: Parcel) : Parcelable {
    @SerializedName("genres")
    var genres: List<Genre?> = emptyList()

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeByte(0x01.toByte())
        dest.writeList(genres)
    }

    companion object {
        val CREATOR: Parcelable.Creator<GenreList?> = object : Parcelable.Creator<GenreList?> {
            override fun createFromParcel(`in`: Parcel): GenreList? {
                return GenreList(`in`)
            }

            override fun newArray(size: Int): Array<GenreList?> {
                return arrayOfNulls(size)
            }
        }
    }

    init {
        if (`in`.readByte().toInt() == 0x01) {
            genres = ArrayList()
            `in`.readList(genres, Genre::class.java.classLoader)
        }
    }
}